glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -75,6 +107,102 @@ open class GleanInternalAPI internal constructor () {
         return initialized.toInt() != 0
     }
 
+    /**
+     * Register the pings generated from `pings.yaml` with Glean.
+     *
+     * @param pings The `Pings` object generated for your library or application
+     * by Glean.
+     */
+    fun registerPings(pings: Any) {
+        // Instantiating the Pings object to send this function is enough to
+        // call the constructor and have it registered in [PingType.pingRegistry].
+        Log.i(LOG_TAG, "Registering pings for ${pings.javaClass.canonicalName}")
+    }
+
+    /**
+     * Enable or disable Glean collection and upload.
+     *
+     * Metric collection is enabled by default.
+     *
+     * When uploading is disabled, metrics aren't recorded at all and no data
+     * is uploaded.
+     *
+     * When disabling, all pending metrics, events and queued pings are cleared.
+     *
+     * When enabling, the core Glean metrics are recreated.
+     *
+     * @param enabled When true, enable metric collection.
+     */
+    fun setUploadEnabled(enabled: Boolean) {
```

Comment:
Do we already have a bug for correctly implementing the logic here?

Reply:
Are you referring to a specific logic currently available in glean-ac? Right now, we don't do much more than that.

Reply:
Weren't there changes how we handle the flipping of the flag, e.g. deleting stored data?

Reply:
Hasn't landed yet :)

Reply:
I see. Let's keep it in mind then for when it lands